packages/grid/src/vaadin-grid-mixin.js

```diff
@@ -855,6 +856,16 @@ export const GridMixin = (superClass) =>
       this._getItem(index, row);
     }
 
+    /** @private */
+    _updateFocusedItem(row) {
+      const focusedItemInRow = row.__virtualIndex === this._focusedItemIndex;
```

Comment:
Don't access `row.__virtualIndex` from grid, since it's a private property used by virtualizer internally. Instead, use `row.index`.